modules/http4s/src/smithy4s/http4s/internals/CompilerContext.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.http4s.internals
+
+import smithy4s.http4s.EntityCompiler
+import smithy4s.http.Metadata
+
+private[http4s] trait CompilerContext[F[_]] {
```

Comment:
This is a helper trait binding the caches together, and also ensuring that the entity cache is consistent (at the type level) with the entity compiler, because we only want to instantiate a single entity cache per router/reverse-router.